addons/mail/static/src/discuss/typing/common/composer_patch.js

```diff
@@ -56,28 +38,20 @@ patch(Composer.prototype, {
                 (command &&
                     (!command.channel_types || command.channel_types.includes(this.thread.type)))
             ) {
-                this.stopTyping();
+                this.threadService.stopTyping(this.thread);
                 return;
             }
         }
-        if (!this.typingNotified && ev.target.value) {
-            this.typingNotified = true;
-            this.notifyIsTyping();
-            browser.setTimeout(() => (this.typingNotified = false), LONG_TYPING);
+        if (!this.thread.typingNotified && ev.target.value) {
+            this.threadService.startTyping(this.thread);
         }
-        this.stopTypingDebounced();
+        this.stopTypingDebounced(this.thread);
     },
     /**
      * @override
      */
     async sendMessage() {
         await super.sendMessage();
-        this.stopTyping();
-    },
-    stopTyping() {
```

Comment:
That's a lot of diff in stable, especially removing methods and such. At least keep the method empty to avoid crashes if it is called by a custo.

Is it possible to keep the method here but properly clean them when the thread changes instead?

Reply:
yes it is, I just feel like putting everything into service would make more sense since it is using thread explicitly. but I agree that this might be too aggressive in stable, I will revert